### DISCUSSION TITLE
Invalid number of years if date is around 6 months ago

### DISCUSSION THREAD

**Discussion by hatkidchan** [moment/moment#6421] [Q&A] [answered]
First of all, my friend on Discord noticed, that my age is suddenly higher than it's supposed to be, it looks like they're using moment.js 'cause that issue also appears here.

**Describe the bug**
If date in past happened 6 or more months ago (say, 1 year and 8 months, 20 years and 6 months), on some specific day, humanized relative time will be off by a year.

**To Reproduce**
```js
const today = new Date();
const past = new Date(
  today.getFullYear() - 20,
  today.getMonth() - 6,
  today.getDate()
);

console.log(moment(past).from(moment(today))); // 21 years
console.log(moment(+past + 86400000).from(moment(today))); // 20 years
console.log(moment.duration({ from: past, to: today }).as('y')); // 20.500919710148178
console.log(moment.duration({ from: +past + 86400000, to: today }).as('y')); // 20.4997235870245
```

**Expected behavior**
Number of years should be number of whole years between dates.

**Moment-specific environment**

* The time zone setting of the machine the code is running on: UTC+0300
* The time and date at which the code was run: `@1663303935` (`Fri Sep 16 07:52:15 AM MSK 2022`)
* Other libraries in use (TypeScript, Immutable.js, etc): none

Please run the following code in your environment and include the output:
```js
console.log((new Date()).toString()) // Fri Sep 16 2022 07:52:49 GMT+0300 (Moscow Standard Time)
console.log((new Date()).toLocaleString()) // 9/16/2022, 7:52:49 AM
console.log((new Date()).getTimezoneOffset()) // -180
console.log(navigator.userAgent) // Mozilla/5.0 (X11; Linux x86_64; rv:104.0) Gecko/20100101 Firefox/104.0
console.log(moment.version) // 2.29.1
```

**Additional context**
I think the issue lies in the https://github.com/moment/moment/blob/000ac1800e620f770f4eb31b5ae908f6167b0ab2/src/lib/duration/humanize.js#L3 and it should be `Math.floor`, not `Math.round`.

#2598 may be relevant too

**Comment by mattjohnsonpint** [comment, marked as answer]
Thanks for the careful analysis! `from()` intentionally produces rounded, humanized text according to relative-time thresholds; it is not an age calculation. For completed calendar years, I'd use `moment(today).diff(moment(past), 'years')`. I'm closing this as answered; friendly rounding and birthdays have different jobs.
